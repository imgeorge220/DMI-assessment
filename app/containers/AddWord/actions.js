/*
 *
 * AddWord actions
 *
 */
import {
  CHANGE_NEW_WORD,
  ADD_WORD,
  ADD_WORD_SUCCESS,
  ADD_WORD_FAILURE,
} from './constants';

export function changeNewWord(newWord) {
  return {
    type: CHANGE_NEW_WORD,
    newWord,
  };
}

export function addWord() {
  return {
    type: ADD_WORD,
  };
}

export function wordAdded() {
  return {
    type: ADD_WORD_SUCCESS,
  };
}

export function wordAddError(error) {
  return {
    type: ADD_WORD_FAILURE,
    error,
  };
}
