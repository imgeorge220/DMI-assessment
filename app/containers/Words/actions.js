/*
 *
 * Words actions
 *
 */
import {
  LOAD_WORDS,
  LOAD_WORDS_SUCCESS,
  LOAD_WORDS_FAILURE,
  ADD_WORD,
  ADD_WORD_SUCCESS,
  ADD_WORD_FAILURE,
} from './constants';

export function loadWords() {
  return {
    type: LOAD_WORDS,
  };
}

export function wordsLoaded(words) {
  return {
    type: LOAD_WORDS_SUCCESS,
    words,
  };
}

export function wordsLoadError(error) {
  return {
    type: LOAD_WORDS_FAILURE,
    error,
  };
}

export function addWord() {
  return {
    type: ADD_WORD,
  };
}

export function wordAdded() {
  return {
    type: ADD_WORD_SUCCESS,
  };
}

export function wordAddError(error) {
  return {
    type: ADD_WORD_FAILURE,
    error,
  };
}
