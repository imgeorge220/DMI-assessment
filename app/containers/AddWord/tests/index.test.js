import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { AddWord, mapDispatchToProps } from '../index';
import { changeNewWord, addWord } from '../actions';
import configureStore from '../../../configureStore';

describe('<AddWord />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddWord dispatch={dispatch} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddWord />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onChangeNewWord', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeNewWord).toBeDefined();
      });

      it('should dispatch changeNewWord when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const newWord = 'test string';
        result.onChangeNewWord({ target: { value: newWord } });
        expect(dispatch).toHaveBeenCalledWith(changeNewWord(newWord));
      });
    });

    describe('onSubmitForm', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onSubmitForm).toBeDefined();
      });

      it('should dispatch addWord and changeNewWord when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onSubmitForm();
        expect(dispatch).toHaveBeenCalledWith(addWord());
        expect(dispatch).toHaveBeenCalledWith(changeNewWord());
      });

      it('should preventDefault if called with event', () => {
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(() => {});
        const evt = { preventDefault };
        result.onSubmitForm(evt);
        expect(preventDefault).toHaveBeenCalledWith();
      });
    });
  });
});
