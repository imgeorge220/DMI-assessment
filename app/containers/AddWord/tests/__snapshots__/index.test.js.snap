// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddWord /> Should render and match the snapshot 1`] = `
<div>
  <form
    class="Form-kd3r69-0 dlxjEv"
  >
    <div>
      <label
        class="Label-sc-1ve2cgc-0 erEqyp"
      >
        Enter a new string
      </label>
      <input
        class="Input-if1qjg-0 daKsjO"
        id="newWord"
        type="textarea"
        value=""
      />
    </div>
    <button
      class="StyledButton-t20ntd-0 gLPxeV"
    >
      Submit
    </button>
  </form>
</div>
`;
